core/survey/_survey/surveyDependencies.js

```diff
@@ -74,7 +76,37 @@ const getNodeDefDependencies = (nodeDefUuid, dependencyType = null) => R.pipe(
   )
 )
 
-const getDeps = (type, nodeDefUuid) => R.pathOr([], [type, nodeDefUuid])
+/**
+ * Returns true if nodeDefUuid is among the dependencies of the specified nodeDefSourceUuid.
+ *
+ */
+const isNodeDefDependentOn = (nodeDefUuid, nodeDefSourceUuid) => survey => {
+  if (nodeDefUuid === nodeDefSourceUuid)
+    return false
+
+  const stack = []
+  stack.push(nodeDefSourceUuid)
+
+  const visitedUuids = new Set()
+
+  while(stack.length > 0) {
+    const nodeDefUuidCurrent = stack.pop()
+
+    if (nodeDefUuid === nodeDefUuidCurrent)
+      return true
+
+    visitedUuids.add(nodeDefUuidCurrent)
+
+    const dependencies = getNodeDefDependencies(nodeDefUuidCurrent)(survey)
+    dependencies.forEach(nodeDefUuidDependent => {
```

Comment:
should normal for-loops be preferred for the codebase in general?

Reply:
yes, I will use Ramda forEach instead, just to be consistent with other parts of the code